### PR TITLE
Enhancement: Add option to select in/out field during import

### PR DESCRIPTION
**Pull request by Jessseee**
While trying to import CSV files from my bank (ING) I ran into the issue that they specify whether a transaction is debit/credit in a separate column. #805 also mentions this issue. I added the option to specify this in/out field during import. I also made a [custom importer](https://github.com/Jessseee/actual-importer) that fixes this issue for now. Check that out incase you also do not feel like editing the CSV files every time you want to import them.

**Usage:**
1. Go to account and import a CSV file.
2. In the import modal, check the new "In/Out" option.
3. Select the correct column name for the "In/Out" CSV field.
4. Input the value of the "In/Out" field that should specifies an "Out" transaction.

### CODE REVIEW

**Comment by MatissJanis** [comment]
👋 Very cool improvement! Is it working though? For me it doesn't seem to be working as expected, but maybe I'm using it incorrectly.

Demo import: 
[ing_example_en.csv](https://github.com/actualbudget/actual/files/13063829/ing_example_en.csv)

Reproduction steps:
1. import the file
2. flip the new "in/out" option "on"
3. change the label to "Credit"
4. observe that both transactions in the preview table are marked as "deposits" (but instead one should be a credit and the other - debit)
5. observe that the "in/out" column in the table has the payee for some reason

**Comment by Jessseee** [comment]
> 5. observe that the "in/out" column in the table has the payee for some reason

@MatissJanis The "in/out" column is not automatically selected so you will have to change the column to "Debit/Credit". I see that I messed up the UX and the default value for selecting the "out value". 

image

File: packages/desktop-client/src/components/modals/ImportTransactions.js [hunk 1]
```diff
@@ -457,11 +500,43 @@ function MultiplierOption({
...
+  return (
+    <View style={{ flexDirection: 'row', gap: 10, height: 28 }}>
+      <CheckboxOption
+        id="form_inout"
+        checked={inOutMode}
+        disabled={disabled}
+        onChange={onToggle}
+      >
+        {inOutMode
+          ? 'in/out field'
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/modals/ImportTransactions.js:519]:
```suggestion
          ? 'in/out identifier value'
```

File: packages/desktop-client/src/components/modals/ImportTransactions.js [hunk 2]
```diff
@@ -457,11 +500,43 @@ function MultiplierOption({
...
+          : 'Select column to specify if amount goes in/out'}
+      </CheckboxOption>
+      {inOutMode && (
+        <Input
+          type="text"
+          value={outValue}
+          onUpdate={onChangeText}
+          placeholder="Out value"
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/modals/ImportTransactions.js:527]:
```suggestion
          placeholder="Value used to tell that this row is a OUT transaction (i.e. Credit)"
```

Open to other suggestions here too. But I definitely think we should work on the original wording to make it more explicit

File: packages/desktop-client/src/components/modals/ImportTransactions.js [hunk 3]
```diff
@@ -757,7 +852,7 @@ export default function ImportTransactions({ modalProps, options }) {
         break;
       }
 
-      let { inflow, outflow, ...finalTransaction } = trans;
+      let { inflow, outflow, inout, ...finalTransaction } = trans;
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/modals/ImportTransactions.js:855]:
```suggestion
      let { inflow, outflow, inOut, ...finalTransaction } = trans;
```

first I read this as a typo of `input`. Took me a second to understand it's actually `in out` :)

File: packages/desktop-client/src/components/modals/ImportTransactions.js [hunk 4]
```diff
@@ -537,16 +612,30 @@ function FieldMappings({
...
-            />
-          </View>
+          <>
+            {inOutMode && (
+              <View style={{ flex: 1 }}>
+                <SubLabel title="In/Out" />
+                <SelectField
+                  options={options}
+                  value={mappings.inout}
+                  onChange={name => onChange('inout', name)}
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/modals/ImportTransactions.js:622]:
```suggestion
                  onChange={name => onChange('inOut', name)}
```

**Review by MatissJanis — CHANGES_REQUESTED** [review]
Overall code looks good! And thanks for explaining how this should work. Lets work on fixing the small nitpicks I left and then I'm happy to merge this.

File: packages/desktop-client/src/components/modals/ImportTransactions.js (hunk 2 above)
Comment (Jessseee) [reply packages/desktop-client/src/components/modals/ImportTransactions.js:527]: Had to make sure the placeholder actually fits the input field. Hope this is clear enough, I think the "i.e. Credit" helps a lot.
image

**Review by MatissJanis — APPROVED** [review]
Thanks for your contribution! Keep em coming :)

**Pull request merged**